Add Share App scenes (optional) to ineligible and eligible signup workflows

EricSiegNW opened the pull request:
This PR accomplishes the following:

1) Connects the APCShareViewController to the APCInEligible signup scene, along with a short label and a button.
2) Hides the back button from the InEligible scene, to discourage the user from simply tapping back and changing their selection criteria.
3) Connects the APCShareViewController in the eligible workflow as an option while waiting to verify email in the APCEmailVerify controller along with a short label and button. This will not be shown on old (3.5") phones due to space issues.

These new buttons/labels to get to the share scene are optional and off by default. Any app can show them by toggling showShareAppInOnboarding to true. The removal of the back option from the APCInEligibleViewController is universal.

EricSiegNW committed:
Initial AppCore changes to add share step when ineligible

EricSiegNW committed:
Merge branch 'master' of https://github.com/EricSiegNW/AppCore into feature/Bridge-566

EricSiegNW committed:
Add missing definitions

EricSiegNW committed:
Update call to align with newest version of ResearchKit

EricSiegNW committed:
Add optional parameter to show share options in onboarding and hide the associated label/button unless showShareAppInOnboarding = YES

EricSiegNW committed:
Hide back button if not eligible, and fix issue with APCShareViewController not popping scene

EricSiegNW committed:
Revert SmartSurveyTask change to keep compatible with RK/Stable instead of RK/Master

Erin-Mounts committed:
Merge pull request #1 from EricSiegNW/feature/Bridge-566

Feature/bridge 566

EricSiegNW committed:
Add Share This App option in verify email

Erin-Mounts committed:
Merge pull request #2 from EricSiegNW/feature/add_share_when_eligible_to_email_verification

Add Share This App option in verify email

EricSiegNW committed:
Merge branch 'master' of https://github.com/ResearchKit/AppCore

Conflicts:
	APCAppCore/APCAppCore/UI/Model/APCOnboarding.m
	APCAppCore/APCAppCore/UI/Onboarding/SignUp/APCInEligibleViewController.m

Merge in ResearchKit/AppCore Changes
Resolve merge issues

Erin-Mounts committed:
Merge pull request #3 from EricSiegNW/master

Merge in changes from RK/Appcore

EricSiegNW committed:
Add Share Scene to APCOnboardingManager

Erin-Mounts committed:
Merge pull request #4 from EricSiegNW/master

Add Share Scene to APCOnboardingManager

p2 commented on APCAppCore/APCAppCore/Startup/APCAppDelegate.h:
```diff
@@ -60,6 +60,7 @@ static NSString*    const kDatabaseName                     = @"db.sqlite";
 @property (strong, nonatomic) APCProfileViewController * profileViewController;
 @property (nonatomic) BOOL disableSignatureInConsent;
 @property (nonatomic, strong) APCDataUploader *dataUploader;
+@property (nonatomic) BOOL showShareAppInOnboarding;
```
I realize you probably started this before the onboarding manager was merged. This setting should be a property of APCOnboardingManager as to not re-introduce dependencies on the app delegate in `APCInEligibleViewController`.

p2 commented on APCAppCore/APCAppCore/UI/Onboarding/SignUp/APCInEligibleViewController.m:
```diff
@@ -36,6 +36,7 @@
 #import "APCOnboardingManager.h"
 #import "APCLog.h"
 #import "APCCustomBackButton.h"
+#import "APCAppDelegate.h"
```
This dependency can be avoided as per my other comment (making `showShareAppInOnboarding` a property of the onboarding manager).

EricSiegNW committed:
Move showShareAppInOnboarding property out of AppDelegate and into APCOnboardingManager

Erin-Mounts committed:
Merge pull request #5 from EricSiegNW/master

Move showShareAppInOnboarding property out of AppDelegate …

EricSiegNW committed:
Merge branch 'master' of https://github.com/ResearchKit/AppCore

Conflicts:
	APCAppCore/APCAppCore/UI/Model/APCOnboardingManager.h

Erin-Mounts committed:
Merge pull request #6 from EricSiegNW/master

Merge in RK/Appcore changes, fix merge conflict

EricSiegNW commented:
share setting moved out of AppDelegate and into APCOnboardingManager per @p2 's suggestion.

EricSiegNW committed:
SBBProfileManager -> SBBUserManager updates, related renames/updates

EricSiegNW committed:
Update schedule importer to new SDK, split survey and non survey tasks

Erin-Mounts committed:
Remove team from APCAppCore target

EricSiegNW committed:
Merge branch 'master' of https://github.com/Sage-Bionetworks/AppCore into feature/update_to_bridge_v3

Erin-Mounts committed:
Merge branch 'master' of https://github.com/ResearchKit/AppCore into Sage-master

EricSiegNW committed:
display subtitles for schedules from server

EricSiegNW committed:
Add task id to view controller mapping check

EricSiegNW committed:
Merge branch 'v3update' of https://github.com/Sage-Bionetworks/AppCore into feature/update_to_bridge_v3

EricSiegNW committed:
remove unnecessary reference

Erin-Mounts committed:
Merge pull request #7 from EricSiegNW/feature/update_to_bridge_v3

Feature/update to bridge v3

Erin-Mounts committed:
Ignore unknown tasks

rather than say crashing, or showing them but not being able to do anything

Erin-Mounts committed:
Merge pull request #8 from Sage-Bionetworks/ignore-new-tasks

Ignore unknown tasks in server schedules

Erin-Mounts committed:
Merge branch 'master' of github.com:ResearchKit/AppCore into sage-master

Erin-Mounts committed:
remove "biological"

Erin-Mounts committed:
Fix bug introduced by recent refactor

specifically https://github.com/ResearchKit/AppCore/commit/869ee294dc15c540a9f945cb5def44c45699c12f
-- check for resultSummary != nil wasn't in previous code, and was preventing results of activity tasks from being stored to core data. As long as there's either a createResultSummaryBlock *or* resultSummary is passed in here, this should be OK.

EricSiegNW committed:
Don't crash the app if the hardcoded json schedules list is empty

Erin-Mounts committed:
Merge pull request #9 from EricSiegNW/feature/Bridge-704

Don't crash the app if the hardcoded json schedules list is empty
